pkg/appmanager/routing.go

```diff
@@ -195,21 +195,13 @@ func processIngressRules(
 	return &rls
 }
 
-func newHttpRedirectPolicyRule(httpsPort int32) *Rule {
-	loc := fmt.Sprintf(`tcl:https://[getfield [HTTP::host] ":" 1]:%d[HTTP::uri]`, httpsPort)
-	redirAction := action{
-		Name:      "0",
-		HttpReply: true,
-		Location:  loc,
-		Redirect:  true,
-		Request:   true,
-	}
-	rule := Rule{
-		Name:    httpRedirectRuleName,
-		Actions: []*action{&redirAction},
-		Ordinal: 0,
-	}
-	return &rule
+func httpRedirectIRule(port string) string {
```

Comment:
port should be passed in as int and %d used in the Sprintf() call rather than forcing the callers to all do the conversion.

Reply:
Yep, good catch. I've gotten %s burned into my mind, I keep forgetting to change types.